generators/entity/index.js

```diff
@@ -1459,7 +1490,11 @@ module.exports = EntityGenerator.extend({
             this.entityClassPlural = pluralize(this.entityClass);
             this.entityInstance = _s.decapitalize(this.name);
             this.entityInstancePlural = pluralize(this.entityInstance);
-            this.entityTableName = _s.underscored(this.name).toLowerCase();
+            if (_.isUndefined(this.tableName)) {
```

Comment:
Why is this not required? If we don't specify the table name with --table-name option, we need to set the @Table entity annotation to the entity name. If we have the table name then this will be the @Table annotation will be set to this name.